WordPress/gutenberg
# Rename section block to group

## Description
Closes #14898

Renames the Section block to Group.

## How has this been tested?
- Manual testing
- Updated E2E tests

See also https://github.com/WordPress/gutenberg/pull/13964#issue-254541645 for further details on testing (but replace the word section with group in your mind).

## Screenshots 
![Screen Shot 2019-04-11 at 4 23 54 pm](https://user-images.githubusercontent.com/677833/55942195-76c48e00-5c76-11e9-87a7-54f5311d39d3.png)

## Types of changes
Breaking change (but one that should be ok if it's merged before the next release)





## Checklist:
